runtime/src/iree-amd-aie/driver/xrt/direct_command_buffer.cc

```diff
@@ -0,0 +1,414 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree-amd-aie/driver/xrt/direct_command_buffer.h"
+
+#include "iree-amd-aie/driver/xrt/native_executable.h"
+#include "iree-amd-aie/driver/xrt/pipeline_layout.h"
+#include "iree-amd-aie/driver/xrt/xrt_buffer.h"
+#include "iree/hal/utils/resource_set.h"
+
+typedef struct iree_hal_xrt_direct_command_buffer_t {
+  iree_hal_command_buffer_t base;
+  iree_allocator_t host_allocator;
+
+  // A resource set to maintain references to all resources used within the
+  // command buffer. Reset on each begin.
+  iree_hal_resource_set_t* resource_set;
+
+  // Staging arena used for host->device transfers.
+  iree_arena_allocator_t arena;
+
+  struct {
+    xrt::bo* bindings[IREE_HAL_XRT_MAX_DESCRIPTOR_SET_BINDING_COUNT];
+    // offset and length are used to get the sub buffer at kernel launch
```

Comment:
Nit: be consistent regarding comments--capitalize the first word and end with `.`.